docs/writing-rules/testing-rules.md

```diff
@@ -31,6 +31,27 @@ Other than annotations there are three things to remember when creating tests:
 The `.test.yaml` file extension can also be used for test files. This is necessary when testing YAML language rules.
 :::
 
+## Testing autofix
+
+Semgrep's testing mechanism also provides a way to test the behaviour of any fixes defined in the rules.
+To define a test for autofix behaviour: 
+
+1. Create a new code file with the `.fixed` suffix before the annotation. 
+   For example, the autofix test file for a rule with test code in `path/to/rule.py` will be named `path/to/rule.fixed.py`.
+2. This file should contain the expected result of applying autofixes to the test code file.
+3. Any future runs of the command `semgrep --test` will apply the autofix to the original testcode (`path/to/rule.py`) and verify whether this matches the expected outcome defined in the autofix test file (`path/to/rule.fixed.py`).
+   If there is a mismatch, the line diffs will be printed.
+
+:::info
+Hint: creating an autofix test for a rule whose autofix is functioning as expected can take less than a minute with the following flow of commands:
```

Comment:
```suggestion
Hint: Creating an autofix test for a rule with autofix can take less than a minute with the following flow of commands:
```

I removed "whose autofix is functioning as expected" because ... aren't we in the process of testing that the autofix is functioning? I hope I am making sense @pieterdc1 .